fleet_vehicle_inspection/models/fleet_vehicle.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 - TODAY, Marcel Savegnago - Escodoo https://www.escodoo.com.br
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class FleetVehicle(models.Model):
+
+    _inherit = 'fleet.vehicle'
+
+    inspection_ids = fields.One2many(
+        'fleet.vehicle.inspection',
+        'vehicle_id',
+        'Inspection Logs'
+    )
+
+    inspection_count = fields.Integer(
+        compute="_compute_inspection_count",
+        string='# Inspection Count')
+
+    @api.depends('inspection_ids')
+    def _compute_inspection_count(self):
+        for rec in self:
+            rec.inspection_count = len(
+                rec.inspection_ids)
+
+    @api.multi
+    def action_view_inspection(self):
+        action = self.env.ref(
+            "fleet_vehicle_inspection.fleet_vehicle_inspection_act_window").read()[0]
+        if self.inspection_count > 1:
+            action["domain"] = [("id", "in", self.inspection_ids.ids)]
+        else:
+            action['context'] = self.env.context.copy()
+            action['context'].update({'default_vehicle_id': self.id})
```

Comment:
I think that `default_vehicle_id` should go in context regardless view type of the action. At this time, when I press the button create from the tree view the vehicle is not automatically set.

Reply:
@mamcode try again please. Thanks.